at_client/lib/src/stream/stream_sender.dart

```diff
@@ -0,0 +1,122 @@
+import 'dart:convert';
+import 'dart:io';
+import 'package:async/async.dart';
+import 'package:at_client/src/client/remote_secondary.dart';
+import 'package:at_client/src/converters/encryption/aes_converter.dart';
+import 'package:at_client/src/service/encryption_service.dart';
+import 'package:at_client/src/stream/at_stream_request.dart';
+import 'package:at_client/src/stream/at_stream_response.dart';
+import 'package:at_client/src/util/network_util.dart';
+import 'package:path/path.dart';
+import 'package:at_utils/at_logger.dart';
+import 'package:at_commons/at_commons.dart';
+
+class StreamSender {
+  String streamId;
+
+  StreamSender(this.streamId);
+
+  late RemoteSecondary remoteSecondary;
+
+  EncryptionService? encryptionService;
+
+  final _logger = AtSignLogger('StreamSender');
+
+  final chunkSize = 1024;
+
+  /// Sends a file specified in [atStreamRequest.filePath] through a stream verb.
+  /// onDone[AtStreamResponse] is called upon stream completion.
+  /// onError[AtStreamResponse] is called when stream errors out.
+  /// If a stream transfer has to be resumed then specify [atStreamRequest.startByte]. Stream transfer will be resumed from startByte.
+  Future<void> send(AtStreamRequest atStreamRequest, Function onDone,
```

Comment:
Can an app send two-stream requests in parallel? i.e calling send for two files.

Reply:
Should the stream id is important for the caller to know before onDone or onError is called?